src/clib/pioc.c

```diff
@@ -595,6 +595,10 @@ PIOc_InitDecomp(int iosysid, int pio_type, int ndims, const int *gdimlen, int ma
     /* Remember the maplen. */
     iodesc->maplen = maplen;
 
+    /* check if the decomp is valid for write or is read-only */
```

Comment:
I'm not sure what to do here.   check_compmap gathers the entire compmap to comp task 0 and checks for duplicate values.  If any are found it's an error to use for write and I've added an assert to the darray_write routines to catch this.  But this is a relatively expensive and somewhat memory intense operation so my inclination is to leave it off.